Fix colortemp conversion for osramlightify

amelchio opened the pull request:
## Description:

**Related issue (if applicable):** fixes #6498

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully.

amelchio committed:
Fix colortemp conversion for osramlightify

Copied from the LIFX fix in 75df4be73391d5255371632f5fb08d7251e85b13.

amelchio committed:
Fix style

armills commented on homeassistant/components/light/osramlightify.py:
```diff
@@ -96,7 +94,7 @@ def __init__(self, light_id, light, update_lights):
         self._rgb = (0, 0, 0)
         self._name = ""
-        self._temperature = TEMP_MIN
+        self._temperature = 2000
```
While we're doing cleanup here, can you just change `self._brightness`, `self._rgb`, `self._name`, `self._temperature`, and `self._state` assignments in `__init__` to `None`? These will get overwritten when `self.update()` is called, so it's safer/cleaner to initialize them to `None` since it shouldn't matter if everything is working.

Also if you're feeling extra ambitious you can drop the `self.update()` call in `__init__`, and change `add_devices_callback(new_lights)` to `add_devices_callback(new_lights, True)` to migrate to the new style and remove IO from the constructor.

amelchio committed:
Updates from review

@armills:

While we're doing cleanup here, can you just change self._brightness,
self._rgb, self._name, self._temperature, and self._state assignments in
__init__ to None? These will get overwritten when self.update() is called, so
it's safer/cleaner to initialize them to None since it shouldn't matter if
everything is working.

amelchio commented:
Sure. I did not like putting the 2000 constant in there in the first place.

However, I do not have the hardware to test my modifications so I am not too ambitious.

armills commented:
Ah, that's fair enough.

I think it's pretty clear at least from a visual inspection that all of those attributes get unconditionally assigned in `update`. I had assumed you had the hardware to double-check. :) It might be a good idea to recruit someone from the forums/chatroom to test it, but it looks pretty safe to me.

emlove reviewed (approved):
Looks good, but it wouldn't hurt to get a sanity check from someone with the hardware.

amelchio commented:
The #6498 issue has a success report on my initial commit.

armills commented:
Ah, didn't see that context. I'm just going to give this some time to let other devs take a look before we merge, since it was just opened.

deisi commented:
When this is merged, I have a follow up PR ready, to add the lightify group feature to HA, similar to how it is handled with hue.

balloob commented:
Thanks 🐬

balloob commented:
cherry-picked
